### PR TITLE
ENH: BundleParc v2

### PR DESCRIPTION
**Pull request by AntoineTheb**
# Quick description

v2 of BundleParc, to be included in the actual paper. Main change is the removal of a WM mask as input.

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

scil_fodf_bundleparc.py ~/.scilpy/tracking/fodf.nii.gz

# Checklist

- [X] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [X] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scilpy/ml/bundleparc/utils.py [hunk 1]
```diff
@@ -2,107 +2,95 @@
...
+    # module.
+    net_state_dict = {'.'.join(k.split('.')[1:]): v for k, v in
+                      state_dict.items() if 'bundleparcnet' in k}
+    model = BundleParcNet(45)
+
+    model.load_state_dict(net_state_dict)
+    model.to(device)
+
+    # Put the model in eval mode to fix dropout and other stuff
+    model.eval()
```
[the remark this answers is not in the document]
Comment (AntoineTheb) [reply scilpy/ml/bundleparc/utils.py:45]: No

File: scripts/scil_fodf_bundleparc.py [hunk 2]
```diff
@@ -97,75 +93,68 @@ def main():
...
-        label_img = resample_volume(label_img, ref_img=wm_in,
-                                    iso_min=False,
-                                    voxel_res=None,
-                                    interp='nn',
-                                    enforce_dimensions=False)
+                                    resampled_img.affine,
+                                    resampled_img.header, dtype=np.uint16)
+
+        # Resampling volume to fit the original image size
+        resampled_label = resample_volume(label_img, ref_img=None,
```
[the remark this answers is not in the document]
Comment (AntoineTheb) [reply scripts/scil_fodf_bundleparc.py]: no

File: scripts/scil_fodf_bundleparc.py [hunk 3]
```diff
@@ -97,75 +92,68 @@ def main():
     parser = _build_arg_parser()
     args = parser.parse_args()
 
-    assert_inputs_exist(parser, [args.in_fodf, args.in_wm])
+    assert_inputs_exist(parser, [args.in_fodf])
     assert_outputs_exist(parser, args, args.out_prefix)
```
Comment (frheault) [inline scripts/scil_fodf_bundleparc.py:96]: I think here you need to generate all of your output (for loop, with all bundles and creates the expected list of output (if using out_folder, skip this, anyway the folder must be empty)

File: scripts/scil_fodf_bundleparc.py [hunk 4]
```diff
@@ -3,17 +3,17 @@
...
 
 Example output:
     sub-001__AF_left.nii.gz, sub-001__AF_right.nii.gz, ..., sub-001__UF_right.nii.gz
 
 The output can be further processed with scil_bundle_mean_std.py to compute statistics for each bundle.
 
-This script requires PyTorch to be installed. To install it, see the official website: https://pytorch.org/get-started/locally/
+This script requires PyTorch==2.2.1 to be installed. See the official website: https://pytorch.org/get-started/locally/
 
 This script requires a GPU with ~6GB of available memory. If you use
```
Comment (frheault) [inline scripts/scil_fodf_bundleparc.py:18]: Does this work with CPU too, my laptop seems to be crashing when using this (but not my desktop).

File: scripts/scil_fodf_bundleparc.py [hunk 5]
```diff
@@ -63,23 +54,27 @@ def _build_arg_parser():
         formatter_class=RawTextHelpFormatter)
 
     parser.add_argument('in_fodf', type=str,
-                        help='fODF input of order 6 or above. If the input '
-                             'has more than 28 coefficients, only the first 28'
-                             ' will be used.')
-    parser.add_argument('in_wm', type=str,
-                        help='Whole-brain WM mask input.')
+                        help='fODF input.')
     parser.add_argument('--out_prefix', type=str, default='',
```
Comment (frheault) [inline scripts/scil_fodf_bundleparc.py:58]: You don't need type=str (default)

File: scripts/scil_fodf_bundleparc.py [hunk 6]
```diff
@@ -63,23 +54,27 @@ def _build_arg_parser():
...
+    parser.add_argument('--keep_biggest_blob', action='store_true',
+                        help='If set, only keep the biggest blob predicted.')
     parser.add_argument('--half_precision', action='store_true',
                         help='Use half precision (float16) for inference. '
                              'This reduces memory usage but may lead to '
                              'reduced accuracy.')
+    parser.add_argument('--bundles', choices=DEFAULT_BUNDLES,
+                        nargs='+', type=str,
+                        help='Bundles to predict. Default is [%(default)s].')
     parser.add_argument('--checkpoint', type=str,
```
Comment (frheault) [inline scripts/scil_fodf_bundleparc.py:78]: type=str (everywhere)

File: scripts/scil_fodf_bundleparc.py [hunk 7]
```diff
@@ -63,23 +54,27 @@ def _build_arg_parser():
         formatter_class=RawTextHelpFormatter)
 
     parser.add_argument('in_fodf', type=str,
```
Comment (frheault) [inline scripts/scil_fodf_bundleparc.py:56]: You don't need type=str (default)

File: scripts/scil_fodf_bundleparc.py [hunk 8]
```diff
@@ -63,23 +54,27 @@ def _build_arg_parser():
...
                         help='Output file prefix. Default is nothing. ')
-    parser.add_argument('--out_folder', type=str, default='',
-                        help='Output destination. Default is the current '
-                             'directory.')
+    parser.add_argument('--out_folder', type=str, default='bundleparc',
+                        help='Output destination. Default is [%(default)s].')
     parser.add_argument('--nb_pts', type=int, default=50,
                         help='Number of divisions per bundle. '
                              'Default is [%(default)s].')
+    parser.add_argument('--min_blob_size', type=int, default=50,
```
Comment (frheault) [inline scripts/scil_fodf_bundleparc.py:65]: Maybe mention this argument make sense for a adult brain in 1x1x1? (here or in docstring)

File: scripts/scil_fodf_bundleparc.py (hunk 4 above)
Comment (AntoineTheb) [reply scripts/scil_fodf_bundleparc.py:18]: Yes, you need to install the right version though. How does it crash ?

**Review by arnaudbore — APPROVED** [review]
